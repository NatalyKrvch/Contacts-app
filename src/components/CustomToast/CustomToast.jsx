import { Toast, ToastContainer } from 'react-bootstrap'

const CustomToast = ({ show, label, message, onClose, bg = 'danger' }) => {
  return (
    <ToastContainer position="top-end" className="p-3">
      <Toast show={show} onClose={onClose} bg={bg} delay={3000} autohide>
        <Toast.Header>
          <strong className="me-auto">{label}</strong>
        </Toast.Header>
        <Toast.Body>{message}</Toast.Body>
      </Toast>
    </ToastContainer>
  )
}

export default CustomToast
