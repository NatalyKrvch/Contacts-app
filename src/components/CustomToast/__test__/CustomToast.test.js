import { render, screen, fireEvent } from '@testing-library/react';
import CustomToast from '../CustomToast';

describe('CustomToast', () => {
  const mockOnClose = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders CustomToast correctly when shown', () => {
    const { container } = render(
      <CustomToast
        show={true}
        label="Test Label"
        message="This is a test message"
        onClose={mockOnClose}
      />
    );

    expect(screen.getByText('Test Label')).toBeInTheDocument();
    expect(screen.getByText('This is a test message')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('calls onClose when close button is clicked', () => {
    render(
      <CustomToast
        show={true}
        label="Test Label"
        message="This is a test message"
        onClose={mockOnClose}
      />
    );

    fireEvent.click(screen.getByRole('button', { name: /close/i }));

    expect(mockOnClose).toHaveBeenCalledTimes(1);
    expect(screen.getByText('Test Label')).toBeInTheDocument();
    expect(screen.getByText('This is a test message')).toBeInTheDocument();
  });

  it('renders with default background color', () => {
    const { container } = render(
      <CustomToast
        show={true}
        label="Test Label"
        message="This is a test message"
        onClose={mockOnClose}
      />
    );

    expect(screen.getByRole('alert')).toHaveClass('bg-danger');
    expect(container).toMatchSnapshot();
  });

  it('renders with custom background color', () => {
    const { container } = render(
      <CustomToast
        show={true}
        label="Test Label"
        message="This is a test message"
        onClose={mockOnClose}
        bg="success"
      />
    );

    expect(screen.getByRole('alert')).toHaveClass('bg-success');
    expect(container).toMatchSnapshot();
  });
});
