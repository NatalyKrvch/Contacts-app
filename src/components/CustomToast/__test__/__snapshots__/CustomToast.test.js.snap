// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomToast renders CustomToast correctly when shown 1`] = `
<div>
  <div
    class="toast-container top-0 end-0 p-3"
  >
    <div
      aria-atomic="true"
      aria-live="assertive"
      class="fade toast bg-danger show"
      role="alert"
    >
      <div
        class="toast-header"
      >
        <strong
          class="me-auto"
        >
          Test Label
        </strong>
        <button
          aria-label="Close"
          class="btn-close"
          data-dismiss="toast"
          type="button"
        />
      </div>
      <div
        class="toast-body"
      >
        This is a test message
      </div>
    </div>
  </div>
</div>
`;

exports[`CustomToast renders with custom background color 1`] = `
<div>
  <div
    class="toast-container top-0 end-0 p-3"
  >
    <div
      aria-atomic="true"
      aria-live="assertive"
      class="fade toast bg-success show"
      role="alert"
    >
      <div
        class="toast-header"
      >
        <strong
          class="me-auto"
        >
          Test Label
        </strong>
        <button
          aria-label="Close"
          class="btn-close"
          data-dismiss="toast"
          type="button"
        />
      </div>
      <div
        class="toast-body"
      >
        This is a test message
      </div>
    </div>
  </div>
</div>
`;

exports[`CustomToast renders with default background color 1`] = `
<div>
  <div
    class="toast-container top-0 end-0 p-3"
  >
    <div
      aria-atomic="true"
      aria-live="assertive"
      class="fade toast bg-danger show"
      role="alert"
    >
      <div
        class="toast-header"
      >
        <strong
          class="me-auto"
        >
          Test Label
        </strong>
        <button
          aria-label="Close"
          class="btn-close"
          data-dismiss="toast"
          type="button"
        />
      </div>
      <div
        class="toast-body"
      >
        This is a test message
      </div>
    </div>
  </div>
</div>
`;
